app/src/main/java/space/taran/arknavigator/mvp/presenter/GalleryPresenter.kt

```diff
@@ -56,7 +56,7 @@ class GalleryPresenter(
         return tags
     }
 
-    fun replaceTags(resource: ResourceId, tags: Tags) {
+    fun replaceTags(resource: ResourceId, tags: Tags) = presenterScope.launch {
         Log.d(GALLERY_SCREEN, "tags $tags added to $resource")
         storage.setTags(resource, tags)
```

Comment:
The same question about `presenterScope`. If this coroutine will be canceled, can it lead to `persist()` call be executed only half-way? If yes, then the storage file might be deleted without new version being written. This might be rare case, but potential outcome is really bad.

Reply:
Yes, `persist()` can be half done if it contains multiple suspend functions inside. Similar to `replaceTags(resource: ResourceId, tags: Tags)` and `storage.setTags(resource, tags)`